### PR TITLE
gnomeExtensions.unite: remove manually packaging and use extension override

### PR DESCRIPTION
**Pull request by rhoriguchi**
###### Motivation for this change

Depends on #140285. Since the current version in `extensions.json` is lower than the packed one. 

Use the new override logic from #137131 to add `buildInputs`. 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by piegamesde** [comment]
Looks great, thank you. Would you mind also adding yourself to the maintainers of that package using the override?

**Comment by rhoriguchi** [comment]
> Looks great, thank you. Would you mind also adding yourself to the maintainers of that package using the override?

Sure, I've added myself.

**Review by piegamesde — APPROVED** [review]

File: pkgs/desktops/gnome/extensions/extensionOverrides.nix [hunk 1]
```diff
@@ -2,6 +2,7 @@
   lib,
   ddcutil,
   gjs,
+  xprop,
 }:
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome/extensions/extensionOverrides.nix:6]:
```suggestion
, lib
, ddcutil
, gjs
, xprop
}:
```

Comment (piegamesde) [reply pkgs/desktops/gnome/extensions/extensionOverrides.nix]: Seriously?

Comment (andir) [reply pkgs/desktops/gnome/extensions/extensionOverrides.nix]: Supersandro considered harmful.

**Pull request merged**